colorfield/fields.py

```diff
@@ -12,28 +12,51 @@
     from django.utils.translation import ugettext_lazy as _
 
 
-COLOR_RE = re.compile('^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$')
-color_validator = RegexValidator(
-    COLOR_RE, _('Enter a valid color.'), 'invalid')
+COLOR_HEX_RE = re.compile('^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$')
+color_hex_validator = RegexValidator(
+    COLOR_HEX_RE, _('Enter a valid color.'), 'invalid')
+
+COLOR_HEXA_RE = re.compile('#([A-Fa-f0-9]{8}|[A-Fa-f0-9]{4})$')
+color_hexa_validator = RegexValidator(
+    COLOR_HEXA_RE, _('Enter a valid color.'), 'invalid')
+
+VALIDATORS_PER_FORMAT = {
+    'hex': color_hex_validator,
+    'hexa': color_hexa_validator
+}
+
+WHITE_IN_FORMAT = {
+    'hex': '#FFFFFF',
+    'hexa': '#FFFFFFFF'
+}
 
 
 class ColorField(models.CharField):
 
-    default_validators = [color_validator]
+    default_validators = []
 
     def __init__(self, *args, **kwargs):
+        color_format = kwargs.pop('color_format', 'hex')
```

Comment:
- I would avoid to repeat the prefix `color_` in options, let's name it just `format`.
- I would also force lowercase, in this way if a dev writes `format='HEX'`, it works anyway.
```python
color_format = color_format.lower()
```

Reply:
✔️